src/test/prelude.rs

```diff
@@ -18,3 +18,15 @@ impl<F: Future + Sized> FutureImmediateExt for F {
             .expect("future did not resolve immediately")
     }
 }
+
+/// An extension future to wait for a future.
+pub trait FutureWaitExt: Future {
```

Comment:
Maybe the `wait` and `immediate` methods can be combined into a `FutureTestExt` trait instead of having two separate traits for testing.